Library/Homebrew/cmd/list.rb

```diff
@@ -36,6 +37,11 @@ class List < AbstractCommand
         switch "--pinned",
                description: "List only pinned formulae, or only the specified (pinned) " \
                             "formulae if <formula> are provided. See also `pin`, `unpin`."
+        switch "--manual", "--installed-on-request",
+               description: "List the formulae installed on request."
+        switch "--auto", "--installed-as-dependency",
+               description: "List the formulae installed automatically."
```

Comment:
```suggestion
               description: "List the formulae only installed indirectly as dependencies."
```

Nit: Wording

Reply:
> since a formula can be both installed-on-request and installed-as-dependency

Speaking of which, I wonder if it makes sense to add another option to list those not installed on request. (The current logic is incorrect for this reason; I'll fix it in the next push.)

Reply:
> I wonder if it makes sense to add another option to list those not installed on request

@ZhongRuoyu Unless you personally need this: no for now, can add later.

Reply:
Whatever wording we have, it should be consistent with:

https://github.com/Homebrew/brew/blob/b5f857b627dc9ae181b91d2d7af906955dfff54e/Library/Homebrew/cmd/leaves.rb#L15-L18